### PR TITLE
Config v2

### PR DESCRIPTION
**Pull request by michaeldjeffrey**
Allows slices of DevAddr under a NetID to go to different places with different settings.

### CODE REVIEW

File: src/pp_config.erl [hunk 1]
```diff
@@ -297,17 +346,17 @@ delete_udp_worker(Pid) ->
     1 = ets:select_delete(?UDP_WORKER_ETS, Spec),
     ok.
 
--spec lookup_udp_workers_for_net_id(NetID :: integer()) -> list(pid()).
-lookup_udp_workers_for_net_id(NetID) ->
-    [P || {_, P} <- ets:lookup(?UDP_WORKER_ETS, NetID)].
+%% -spec lookup_udp_workers_for_net_id(NetID :: integer()) -> list(pid()).
+%% lookup_udp_workers_for_net_id(NetID) ->
```
Comment (dpezely) [inline src/pp_config.erl:350]: Same: `TODO` or remove, please.

File: src/pp_config_v2.erl [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+                            name = Name,
+                            net_id = NetID,
+                            protocol = Protocol,
+                            multi_buy = MultiBuy,
+                            buying_active = BuyingActive,
+                            addr = {range, hex_to_num(Lower), hex_to_num(Upper)}
+                        }
+                    end,
+                    case DevAddrs of
+                        [] -> [#{<<"lower">> => <<"0x00000000">>, <<"upper">> => <<"0xFFFFFFFF">>}];
```
Comment (dpezely) [inline src/pp_config_v2.erl:144]: The 0x Binary values aren't Erlang hex.  If that's intended, please add a comment explaining.

(I see that you have `hex_to_num()` below where you parse the `<<"0x">>` prefix, but on first read here, it looks like a possible typo.)

**Review by dpezely — APPROVED** [review]
Some minor nitpicks that can all be fixed by peppering a few more comments, but it's not dire enough to prevent merging the PR.  Instead, the next person will `git blame` and get to annoy you.  :-)

File: src/pp_config.erl [hunk 3]
```diff
@@ -597,91 +592,65 @@ write_config_to_ets(Config) ->
...
-    try erlang:binary_to_integer(Bin, 16) of
-        Num -> Num
-    catch
-        error:_ ->
-            lager:warning("value is not hex: ~p", [Bin]),
-            Bin
-    end.
-%% clean_base16(_) -> throw(malformed_base16).
+%% -spec write_config_to_disk(string()) -> ok.
+%% write_config_to_disk(Filename) ->
```
[the remark this answers is not in the document]
Comment (michaeldjeffrey) [reply src/pp_config.erl:596]: The next line contains a relevant `FIXME`

**Review by dpezely — APPROVED** [review]

File: src/pp_config_v2.erl (hunk 2 above)
Comment (michaeldjeffrey) [reply src/pp_config_v2.erl:144]: I'll add a good note when I update the docs.
All the "hex" values in packet purchaser right now are roaming1.1 spec "hex" values.

**Pull request merged**